features/plugin/plugin/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/src/config.h.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/config.h" @ONLY)
+
+include(ZeekPlugin)
+
+zeek_plugin_begin(@NS@ @NAME@)
+
+file(GLOB cc_files RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "src/*.cc")
+foreach(file ${cc_files})
+    zeek_plugin_cc(${file})
+endforeach ()
+
+file(GLOB bif_files RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "src/*.bif")
+foreach(file ${bif_files})
+    zeek_plugin_bif(${file})
+endforeach ()
+
+# Copy the dist files to package with the tarball into the top-level
+# build directory to fulfill zeek-plugin-create-package.sh assumptions
+# about them being located in the parent directory.
+set(dist_files README CHANGES COPYING VERSION)
+foreach(file ${dist_files})
+    if ( EXISTS ../${file} )
+        file(COPY ../${file} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/../)
+    endif()
+endforeach ()
```

Comment:
Like I mentioned offline, not a fan of the magic modifications to the "file names" since this makes it pretty hard to modify for users, what if they want to e.g., included a generated file in the tarball? I'd prefer just directly referencing files with relative paths and requiring them to exist; in addition to the technical implementation comment we also need a user-targeted comment explaining what this variable is for -- for anything else we use magic globs 🤢.

Probably worse, I just noticed that packaging already copies some files around, https://github.com/zeek/cmake/blob/2df3b8e82a843b7b8187963d259d32a9fb42b873/ConfigurePackaging.cmake#L133-L140, i.e., we would already create a `README.txt` and `COPYING.txt` in the build directory. This could potentially clash with what users set up here (e.g., if they named their `README` `README.txt`). I do not think there is a good automagic way to fix this, can we augment `zeek-plugin-create-package.sh` so it can understand explicit dist files?

Reply:
> can we augment zeek-plugin-create-package.sh so it can understand explicit dist files?

Not sure I fully understand how to do this, but agree the file copying isn't great - it just attempts to please `zeek-plugin-create-package.sh`. Plainly using `.../VERSION` currently fails as the create package script then copies it not into `dist`, but `dist/../VERSION`. Hmm, we could possibly just strip the leading `../`'s before determining the destination directory within `./dist`, that could actually be "okay".

Reply:
We could also just have the semantics that the files end up in the same place they were relative to the `CMakeLists.txt` with `project` here, e.g., from a subdirectory `../README` (which is the same as `<project>/README`) gets copied `<dist>/README`. I left a similar comment on zeek/cmake#116.